utils/crawl_github_files.py

```diff
@@ -135,21 +135,77 @@ def should_include_file(file_path: str, file_name: str) -> bool:
     owner = path_parts[0]
     repo = path_parts[1]
     
-    # Check if URL contains a specific branch/commit
-    if 'tree' in path_parts:
-        tree_index = path_parts.index('tree')
-        ref = path_parts[tree_index + 1]
-        # Combine all parts after the ref as the path
-        path_start = tree_index + 2
-        specific_path = '/'.join(path_parts[path_start:]) if path_start < len(path_parts) else ""
-    else:
-        ref = "main"  # Default branch
-        specific_path = ""
-    
     # Setup for GitHub API
     headers = {"Accept": "application/vnd.github.v3+json"}
     if token:
         headers["Authorization"] = f"token {token}"
+
+    def fetch_branches(owner: str, repo: str):
+        """Get brancshes of the repository"""
+
+        url = f"https://api.github.com/repos/{owner}/{repo}/branches"
+        response = requests.get(url, headers=headers)
+
+        if response.status_code == 404:
+            if not token:
+                print(f"Error 404: Repository not found or is private.\n"
+                      f"If this is a private repository, please provide a valid GitHub token via the 'token' argument or set the GITHUB_TOKEN environment variable.")
+            else:
+                print(f"Error 404: Path '{path}' not found in repository or insufficient permissions with the provided token.\n"
```

Comment:
The variable `path` doesn't seem to be defined within the scope of the fetch_branches function. 
Should it be `{owner}/{repo}`?